### ISSUE TITLE
Logic module as global

### ISSUE THREAD

**Issue by jh3ady** [jeffbski/react-boilerplate-logic#3]
Hello,

How can we create a logic module as global in our application ? Like for authentication, etc.

Best regards

**Comment by jeffbski** [comment]
@JeanDenisVIDOT thanks for reaching out. You can add global logic and also any injected helpers (to use inside any logic) where we create the original logicMiddleware.

https://github.com/jeffbski/react-boilerplate-logic/blob/master/app/store.js#L15

Original code is:
```js
const injectedHelpers = { requestUtil };
const logicMiddleware = createLogicMiddleware([], injectedHelpers);
```

So you can change it so that you can add in any helpers you want available in all logic and if you want any logic to always be loaded then you can add that here too.

```js
import foo from './foo'; // another helper I want to use in my logic
import globalLogic from './global-logic'; // array of logic to add always
const injectedHelpers = { requestUtil, foo }; // adding foo as another injected helper available to all logic
const logicMiddleware = createLogicMiddleware(globalLogic, injectedHelpers);
```

So by adding additional injected helpers you can use them in your logic like

```js
const abcLogic = createLogic({
  type: CAT,
  transform({ getState, action, requestUtil, foo }, next) {
    // your code
  },
  process({ getState, action, requestUtil, foo }, dispatch, done) {
    // your code
  }
});
```

Then of course create a file globalLogic.js which exports the logic array as the default.

```js
import { createLogic } from 'redux-logic';
const logic1 = createLogic({...});
const logic2 = createLogic({...});
export default [
  logic1,
  logic2
];
```

**Comment by jh3ady** [comment]
@jeffbski thanks for your answer.

So, for instance : 

If I dispatch an action when store has been initialized ( like store.dispatch({ type: 'foo' }) ) I'll be able to execute my logic ?

Is it possible for a logic to listen more than one action for type property ?
Is it possible for a logic to listen more than one action for cancelType property ?

Sorry for my beginner questions but I'm trying to understand.

**Comment by jeffbski** [comment]
Yes, whenever you dispatch after the logic has been loaded then it will be listening and will respond calling your logic functions.

Yes, you can listen to more than one type per logic. The type can be a string, a regex, or an array of strings and regex so you can easily match many types of actions in one logic.

```js
type: [FOO, BAR, /cat.*/]
```

Yes, same thing for cancelType.

No worries on the questions. I am happy to help as I can.
